packages/nocodb/src/lib/meta/helpers/NcPluginMgrv2.ts

```diff
@@ -97,25 +98,56 @@ class NcPluginMgrv2 {
           pluginConfig.id
         );
       }
+    }
+    await this.initPluginsFromEnv();
+  }
 
-      /* init only the active plugins */
-      // if (pluginConfig?.active) {
-      //   const tempPlugin = new plugin.builder(this.app, plugin);
-      //
-      //   this.activePlugins.push(tempPlugin);
-      //
-      //   if (pluginConfig?.input) {
-      //     pluginConfig.input = JSON.parse(pluginConfig.input);
-      //   }
-      //
-      //   try {
-      //     await tempPlugin.init(pluginConfig?.input);
-      //   } catch (e) {
-      //     console.log(
-      //       `Plugin(${plugin?.title}) initialization failed : ${e.message}`
-      //     );
-      //   }
-      // }
+  private static async initPluginsFromEnv() {
+    /*
+     * NC_S3_BUCKET_NAME
+     * NC_S3_REGION
+     * NC_S3_ACCESS_KEY
+     * NC_S3_ACCESS_SECRET
+     * */
+
+    if (
+      process.env.NC_S3_BUCKET_NAME &&
+      process.env.NC_S3_REGION &&
+      process.env.NC_S3_ACCESS_KEY &&
+      process.env.NC_S3_ACCESS_SECRET
+    ) {
+      const s3Plugin = await Plugin.getPluginByTitle(S3PluginConfig.title);
+      await Plugin.update(s3Plugin.id, {
+        active: true,
+        input: JSON.stringify({
+          bucket: process.env.NC_S3_BUCKET_NAME,
+          region: process.env.NC_S3_REGION,
+          access_key: process.env.NC_S3_ACCESS_KEY,
+          access_secret: process.env.NC_S3_ACCESS_SECRET
+        })
+      });
+    }
+
+    if (
+      process.env.NC_SMTP_FROM &&
+      process.env.NC_SMTP_HOST &&
+      process.env.NC_SMTP_PORT &&
+      process.env.NC_SMTP_USERNAME &&
+      process.env.NC_SMTP_PASSWORD
```

Comment:
These are optional on smtp plugin, we can make it optional on env variables too.

Reply:
> `NC_ADMIN_PASSWORD` allows any length of string but sign in form checks if the password is at least 8 characters long so if user provides a password shorter than 8 chars, super user password get updated but you can't login as the sign in form doesn't validate for correct password. We can remove validation from sign in or let users know this behavior.
> 
> Other than that LGTM.

Yes, we can remove validation from the sign-in form and just keep the required validation.